### DISCUSSION TITLE
After Training, upload to HuggingFace

### DISCUSSION THREAD

**Discussion by the3d4nk** [zetavg/LLaMA-LoRA-Tuner#24] [General]
After successfully training, how would I use the LLaMa based model in HuggingFace? I pushed the contents of the `lora_models` folder which I uniquely labeled but it is apparently missing the base model in order to successfully use an inference API?

**Comment by zetavg** [comment]
To my knowledge, the inference API does not support adapter models. You might need to use `.merge_and_unload` to merge the LoRA model to the base model and upload the merged model.

```py
model = AutoModelForCausalLM.from_pretrained(
    base_model_name_or_path, device_map='auto')

model = PeftModel.from_pretrained(
    model,
    lora_model_name_or_path,
    device_map='auto'
)

model = model.merge_and_unload()  # Needs peft>=0.3.0

model.push_to_hub(model_name)

# For the inference API to work, we need to push the tokenizer too.
tokenizer.push_to_hub(model_name)
```

**Comment by the3d4nk** [comment]
This is very helpful. Thank you!
